crates/build/src/args.rs

```diff
@@ -78,9 +79,17 @@ pub enum Network {
 }
 
 impl Network {
+    /// Returns the current network status
+    pub fn network_status(&self) -> Self {
+        match check(None) {
+            Ok(_) => Self::Online,
+            _ => Self::Offline
+        }
+    }
```

Comment:
No we don't want to do this because it will override the user provided `--offline` flag.